timemachine/potentials/nonbonded.py

```diff
@@ -98,7 +97,13 @@ def nonbonded_block_unsummed(xi, xj, box, params_i, params_j, beta, cutoff):
     ri = jnp.expand_dims(xi, axis=1)
     rj = jnp.expand_dims(xj, axis=0)
 
-    dij = jnp.linalg.norm(delta_r(ri, rj, box), axis=-1)
+    w_i = jnp.expand_dims(params_i[:, 3], axis=1)
+    w_j = jnp.expand_dims(params_j[:, 3], axis=0)
+
+    d2ij = delta_r(ri, rj, box)
+    d2ij = jnp.concatenate([d2ij, jnp.array(w_i - w_j).reshape(*d2ij.shape[:-1], 1)], axis=-1)
```

Comment:
The lack of support for w could result in some cases failing due to lambda not being 0.0